### PR TITLE
Sort key CSV prototype

### PR DESCRIPTION
**Pull request by zerebubuth**
Adds functionality to load a CSV file containing lines, each of which is a set of filters plus a target value in the last (rightmost) column. The first line for which all the filters match on a feature will set the target value as a property.

@rmarianski could you review, please?

### CODE REVIEW

File: TileStache/Goodies/VecTiles/transform.py [hunk 1]
```diff
@@ -303,7 +304,7 @@ def road_sort_key(shape, properties, fid, zoom):
     ne_type = properties.get('type', '')
 
     is_railway = railway in (
-        'rail', 'tram', 'light_rail', 'narrow_guage', 'monorail')
+        'rail', 'tram', 'light_rail', 'narrow_gauge', 'monorail')
```
Comment (rmarianski) [inline TileStache/Goodies/VecTiles/transform.py:307]: Nice catch! :zap:

File: TileStache/Goodies/VecTiles/transform.py [hunk 2]
```diff
@@ -3093,3 +2972,155 @@ def normalize_medical_kind(shape, properties, fid, zoom):
...
+class _SetMatcher(object):
+    def __init__(self, values):
+        self.values = values
+
+    def match(self, other):
+        return other in self.values
+
+    def __repr__(self):
+        return repr(self.value)
+
```
Comment (rmarianski) [inline TileStache/Goodies/VecTiles/transform.py:3021]: We could transition the `match` function to be `__call__`, but we can do that later if we wanted.

File: TileStache/Goodies/VecTiles/transform.py [hunk 3]
```diff
@@ -3093,3 +2972,155 @@ def normalize_medical_kind(shape, properties, fid, zoom):
...
+        return None
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    def _type_cast(v):
+        if target_value_type == 'int':
+            return int(v)
+        return v
```
Comment (rmarianski) [inline TileStache/Goodies/VecTiles/transform.py:3118]: Do you think we'll care about having other types in the future? I guess we can update it at that point if we do.

**Comment by rmarianski** [comment]
:+1:

File: TileStache/Goodies/VecTiles/transform.py (hunk 3 above)
Comment (zerebubuth) [inline TileStache/Goodies/VecTiles/transform.py:3118]: Yeah, it'll do `str` and `int` now, and I couldn't think of any others we needed right now. We don't really have many `float` properties.

**Pull request merged**